version: "3.8"

services:
  db:
    image: postgres:16.1
    restart: on-failure
    environment:
      POSTGRES_USER: ${DATABASE_USER}
      POSTGRES_PASSWORD: ${DATABASE_PASSWORD}
      POSTGRES_DB: ${DATABASE_NAME}
    volumes:
      - app_data:/var/lib/postgresql/data
    ports:
      - "${DATABASE_PORT:-5432}:${DATABASE_PORT:-5432}"
    healthcheck:
      test: "/usr/bin/pg_isready --dbname=${DATABASE_NAME} --username=${DATABASE_USER} || exit 1"
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  app_data: {}
